projects/common/src/utilities/formatters/string/highlight.pipe.ts

```diff
@@ -1,20 +1,41 @@
 import { Pipe, PipeTransform } from '@angular/core';
+import { isArray } from 'lodash-es';
 import { assertUnreachable } from '../../lang/lang-utils';
 
 @Pipe({ name: 'htHighlight' })
 export class HighlightPipe implements PipeTransform {
   public transform(
     fullText: string,
-    highlightedText: string | undefined,
+    highlightSnippets: string | undefined | TextHighlightConfig[],
     highlightType: HighlightType = 'mark'
   ): string {
+    if (isArray(highlightSnippets)) {
+      let textWithHighlight = fullText;
+      highlightSnippets.forEach(highlightConfig => {
+        const highlightHtmlTag = getHtmlTagForHighlightType(highlightConfig.highlightType);
+        textWithHighlight = textWithHighlight.replace(
+          highlightConfig.text,
+          `<${highlightHtmlTag}>$&</${highlightHtmlTag}>`
+        );
+      });
+
+      return textWithHighlight;
+    }
+
     const htmlTag = getHtmlTagForHighlightType(highlightType);
 
-    return highlightedText === undefined ? fullText : fullText.replace(highlightedText, `<${htmlTag}>$&</${htmlTag}>`);
+    return highlightSnippets === undefined
```

Comment:
if i called the highlight pipe with no args, I'd expect the default case would be to highlight the whole thing.